app/src/main/java/mod/hilal/saif/components/ComponentsHandler.java

```diff
@@ -574,4 +574,28 @@ private static ArrayList<HashMap<String, Object>> readCustomComponents() {
     public static void refreshCachedCustomComponents() {
         cachedCustomComponents = readCustomComponents();
     }
+
+    public static boolean isValidComponent(HashMap<String, Object> map) {
```

Comment:
when using Java collections, don't make unnecessary constraints like the parameter absolutely having to be a `HashMap`, and in this case, use the interface `Map` instead